exporter/datadogexporter/internal/testutil/logs.go

```diff
@@ -28,20 +32,32 @@ func (jsonLogs *JSONLogs) HasDDTag(ddtags string) bool {
 type DatadogLogsServer struct {
 	*httptest.Server
 	// LogsData is the array of json requests sent to datadog backend
-	LogsData JSONLogs
+	LogsData             JSONLogs
+	connectivityCheck    sync.Once
+	mockNetworkError     sync.Once
+	logsAgentDoneChannel chan bool
 }
 
 // DatadogLogServerMock mocks a Datadog Logs Intake backend server
-func DatadogLogServerMock(overwriteHandlerFuncs ...OverwriteHandleFunc) *DatadogLogsServer {
+func DatadogLogServerMock(logsAgentDoneChannel chan bool, retry bool, overwriteHandlerFuncs ...OverwriteHandleFunc) *DatadogLogsServer {
```

Comment:
It seems like we can just use an appropriate `OverwriteHandleFunc` for the retry instead of adding a new option, can't we?